dace/transformation/passes/simplification/control_flow_raising.py

```diff
@@ -178,21 +212,95 @@ def _lift_conditionals(self, sdfg: SDFG) -> int:
             sdfg.root_sdfg.using_explicit_control_flow = True
         return lifted
 
+    def _lift_unstructured(self, sdfg: SDFG) -> int:
+        """
+        Lift regions of unstructured control flow.
+        When this is called, it is assumed that loops have already been lifted. This implies that any remaining
+        cycles represent unstructured control flow.
+
+        :param sdfg: The SDFG in which to lift unstructured control flow
+        :returns: The number of unstructured control flow blocks lifted
+        """
+        lifted = 0
+        for cfg in sdfg.all_control_flow_regions():
+            if isinstance(cfg, (UnstructuredControlFlow, ConditionalBlock)):
+                continue
+
+            if not cfg.has_cycles():
+                # No cycles, no unstructured control flow.
+                continue
+
+            # Compute immediate dominators
+            idom: Dict[ControlFlowBlock, ControlFlowBlock] = nx.immediate_dominators(cfg.nx, cfg.start_block)
+
+            # Find back edges (u, v) where v dominates u
+            back_edges = []
```

Comment:
Would it make sense to use `cfg_analysis.back_edges` here?

Reply:
True, changed